apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/SampleQuestionDemo.tsx

```diff
@@ -0,0 +1,366 @@
+            grade={grade}
+            answerType={variant.answerType}
+            onSelectOption={handleSelectOption}
+          />
+        )}
+      </CardBody>
+      <CardFooter>
+        <div className="d-flex flex-wrap justify-content-end align-items-center gap-2">
+          <p className="my-0">
+            <i>Answer: {answerText}</i>
+          </p>
+          <div className="flex-grow-1"></div>
+          <div className="d-flex align-items-center gap-2">
+            <Button onClick={handleGenerateNewVariant}>
+              <span className="text-nowrap">New variant</span>
+            </Button>
+
+            <Button onClick={handleGrade}>Grade</Button>
+          </div>
+        </div>
+      </CardFooter>
+    </Card>
+  );
+}
+
+function FeedbackBadge({ grade }: { grade: number }) {
+  const badgeType = run(() => {
+    if (grade === 100) {
+      return 'bg-success';
+    } else if (grade > 0) {
+      return 'bg-warning';
+    } else {
+      return 'bg-danger';
+    }
+  });
+  return (
+    <span className={`badge ${badgeType}`}>
+      {Math.floor(grade)}
+      {'%'}
+    </span>
+  );
+}
+
+function NumericOrStringInput({
+  userInputResponse,
+  placeholder,
+  grade,
+  answerLabel,
+  answerUnits,
+  onChange,
+}: {
+  userInputResponse: string;
+  placeholder: string;
+  grade: number | null;
+  answerLabel: string;
+  answerUnits?: string;
+  onChange: (text: string) => void;
+}) {
+  return (
+    <InputGroup>
+      <InputGroupText key={answerLabel} id="answer-label">
+        {answerLabel}
+      </InputGroupText>
+      <FormControl
+        value={userInputResponse}
+        type="text"
+        aria-label="Sample question response"
+        aria-describedby="answer-label"
+        placeholder={placeholder}
+        onChange={(e) => onChange(e.currentTarget.value)}
+      />
+      {(answerUnits || grade !== null) && (
+        <InputGroupText>
+          <span className={grade !== null ? 'me-2' : ''}>{answerUnits}</span>
+          {grade !== null ? <FeedbackBadge grade={grade} /> : <></>}
+        </InputGroupText>
+      )}
+    </InputGroup>
+  );
+}
+
+function CheckboxOrRadioInput({
+  selectedOptions,
+  options,
+  grade,
+  answerType,
+  onSelectOption,
+}: {
+  selectedOptions: Set<string>;
+  options: VariantOption[];
+  grade: number | null;
+  answerType: 'checkbox' | 'radio';
+  onSelectOption: (option: string) => void;
+}) {
+  return (
+    <div>
+      {options.map((option, index) => (
+        <FormCheck
+          id={`check-${index}`}
+          key={index}
+          type={answerType as 'checkbox' | 'radio'}
+          label={variantOptionToString(option)}
+          value={option.value}
+          checked={selectedOptions.has(option.value)}
+          onChange={() => onSelectOption(option.value)}
+        />
+      ))}
+      {grade !== null ? (
+        <div className="mt-2">
+          <FeedbackBadge grade={grade} />
+        </div>
+      ) : (
+        <></>
```

Comment:
```suggestion
        null
```

That said: `grade != null && ( ... )` is a more conventional way to do conditional rendering in React.

Reply:
I implemented the suggested change and adjusted this code to use `grade !== null && (....)`:

https://github.com/PrairieLearn/PrairieLearn/blob/7b7bca8f608244862435f8df983c8a4d991cdba8/apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/SampleQuestionDemo.tsx#L348-L352

I implemented a similar change for the `FeedbackBadge` in `NumericOrStringInput` as well:

https://github.com/PrairieLearn/PrairieLearn/blob/7b7bca8f608244862435f8df983c8a4d991cdba8/apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/SampleQuestionDemo.tsx#L315